internal/controller/account_resolver.go

```diff
@@ -49,6 +52,69 @@ func (this authGwErrorResponse) String() string {
 	return b.String()
 }
 
+type ExpirableCachedAccountIdResolver struct {
+	AccountIdResolver
+	cache 		*lru.Cache
+	cacheTTL 	time.Duration
+	errorTTL 	time.Duration
+}
+
+
+func NewExpirableCachedAccountIdResolver(baseResolver AccountIdResolver, cacheSize int, cacheTTL, errorTTL time.Duration)(AccountIdResolver, error) {
+	cache, err := lru.New(cacheSize)
+	if err != nil {
+		return nil, err
+	}
+
+	return &ExpirableCachedAccountIdResolver{
+		AccountIdResolver: 	baseResolver,
+		cache:				cache,
+		cacheTTL: 			cacheTTL,
+		errorTTL: 			errorTTL,
+	},nil
+}
+
+func (ecar *ExpirableCachedAccountIdResolver) MapClientIdToAccountId(ctx context.Context, clientID domain.ClientID) (domain.Identity, domain.AccountID, domain.OrgID, error){
+	//Check cache 
+	cached, ok := ecar.cache.Get(clientID)
+	if ok {
+		cachedResult := cached.(cachedResult)
+
+		now := time.Now()
+		//Check if cached result is still valid 
+		if cachedResult.err == nil && now.Sub(cachedResult.timestamp) < ecar.cacheTTL {
+			return cachedResult.identity, cachedResult.accountID, cachedResult.orgID, nil
+		}
+
+		if now.Sub(cachedResult.timestamp) < ecar.errorTTL && cachedResult.err != nil {
+			//if cach err is whin the erro ttl return it
```

Comment:
```suggestion
			//if cache error is within the error ttl return it
```
Drive by typo correction :D